prow/cmd/deck/static/pr-script.js

```diff
@@ -969,19 +1032,77 @@ function createPRCard(pr, builds = [], queries = [], tidePools = []) {
         });
         score = (labels.length + missingLabels.length > 0) ? score
             / (labels.length + missingLabels.length) : 1.0;
-        processedQuery.push(
-            {score: score, labels: labels, missingLabels: missingLabels});
+        processedQueries.push(
+            {score: score, labels: labels, missingLabels: missingLabels, excludedBranches:query.excludedBranches,
+                includedBranches: query.includedBranches, milestone:query.milestone});
     });
     // Sort queries by descending score order.
-    processedQuery.sort((q1, q2) => {
+    processedQueries.sort((q1, q2) => {
+        let q1Excluded = 0, q2Excluded = 0;
+        if (pr.BaseRef && pr.BaseRef.Name &&
+            q1.excludedBranches && q1.excludedBranches.indexOf(pr.BaseRef.Name) !== -1) {
+            q1Excluded = 1;
+        }
+        if (pr.BaseRef && pr.BaseRef.Name &&
+            q2.excludedBranches && q2.excludedBranches.indexOf(pr.BaseRef.Name) !== -1) {
+            q2Excluded = -1;
+        }
+        if (q1Excluded + q2Excluded !== 0){
+            return q1Excluded + q2Excluded;
+        }
+
+        let q1Included = 0, q2Included = 0;
+        if (pr.BaseRef && pr.BaseRef.Name &&
+            q1.includedBranches && q1.includedBranches.indexOf(pr.BaseRef.Name) !== -1) {
+            q1Included = -1;
+        }
+        if (pr.BaseRef && pr.BaseRef.Name &&
+            q2.includedBranches && q2.includedBranches.indexOf(pr.BaseRef.Name) !== -1) {
+            q2Included = 1;
+        }
+        if(q1Included + q2Included !== 0){
+            return q1Included + q2Included;
+        }
+        if(q1.Milestone !== q2.Milestone) {
+            if (pr.Milestone && pr.Milestone.Title && q1.milestone && pr.Milestone.Title === q1.milestone) {
```

Comment:
You have `q1.milestone` on this line and `q1.Milestone` on the previous. One of them must be incorrect?

Also lets just check `pr.Milestone && pr.Milestone.Title` only once.

Reply:
The outer `q1.Milestone !== q2.Milestone` is only to enter inner logic, else the `pr.Milestone` comparison is irrelevant.